provisioning/setup.py

```diff
@@ -83,14 +101,214 @@ def __call__(self, parser, namespace, values, option_string=None):
+                    # Below is snippet of sample nodeinfo from hctl status
+                    # "nodes": [
+                    #    [
+                    #      "ssc-vm-c-0xxx.colo.seagate.com",
+                    #      [
+                    #        [
+                    #          "hax",
+                    #          "0x7200000000000001:0x6",
+                    #          "192.168.9.xxx@tcp:12345:1:1",
+                    #          "started"
+                    #        ],
+                    #        [
+                    #          "confd",
+                    #          "0x7200000000000001:0x9",
+                    #          "192.168.9.xxx@tcp:12345:2:1",
+                    #          "started"
+                    for node_service in nodeinfo[1]:
+                        if node_service[0] == 'hax' and node_service[3] == 'started':
+                            hax_found = True
+                            continue
+                        if node_service[0] == 'confd' and node_service[3] == 'started':
+                            confd_p = True
+                            continue
+                        if node_service[0] == 'ioservice' and node_service[3] == 'started':
+                            ios_count += 1
+                            continue
+                        # Status of 'm0_client' is currently shown as 'unknown' so adding check to avoid failure
+                        if node_service[0] == 'm0_client' and (node_service[3] == 'started' or node_service[3] == 'unknown'):
+                            m0client_count += 1
+
+                    if confd_p_cdf == True:
+                        if confd_p == False:
+                            logging.error('confd service not found in hctl status')
+                            return -1
+                    if hax_found == False:
+                        logging.error('Hax service not found in hctl status')
+                        return -1
+                    if ios_count != ios_count_cdf:
+                        logging.error(str(ios_count) + ' io services were found whereas CDF suggest we should have ' + str(ios_count_cdf))
+                        return -1
+                    if m0client_count != m0client_count_cdf:
+                        logging.error(str(m0client_count) + ' io services were found whereas CDF suggest we should have ' + str(m0client_count_cdf))
+                        return -1
+                    logging.info('Cluster status verified successfully')
+                    return 0
+    logging.error('Failed to find node info in hctl status')
+    return -1
+
 
 def main(argv=None):
     p = argparse.ArgumentParser(description='Configure hare settings')
     p.add_argument('--report-unavailable-features',
                    nargs=0,
                    help='Report unsupported features according to setup type',
                    action=UnsupportedFeatures)
+    p.add_argument('--post_install',
```

Comment:
I did think that it would be good to merge setup.py with hax regular sources because it is possible provisioning/setup.py may grow further. Testing can be done by running setup.py utility instead of functions, but it makes sense to modularize it but I think we can take it up as a separate task.

Reply:
Created #1464. No need to handle this suggestion in scope of this PR.